numpyro/distributions/continuous.py

```diff
@@ -290,6 +291,62 @@ def infer_shapes(concentration):
         return batch_shape, event_shape
 
 
+class EulerMaruyama(Distribution):
+    arg_constraints = {"t": constraints.ordered_vector}
+
+    support = constraints.real
+
+    def __init__(self, t, sde_fn, sde_pars, init_dist, validate_args=None):
+        # we use t rather than (dt, num_steps) to make the api simpler and more general,
+        # `init_dist` will define prior for the init value,
+        # event dimensions will be decided by `init_dist`
+        self.t = t
+        self.dt = jnp.diff(t)
+        self.sde_fn = sde_fn
+        self.sde_pars = sde_pars
+        self.init_dist = init_dist
+
+        batch_shape, event_shape = init_dist.batch_shape, jnp.shape(t)
+
+        super(EulerMaruyama, self).__init__(
+            batch_shape, event_shape, validate_args=validate_args
+        )
+
+    def sample(self, key, sample_shape=()):
+        assert is_prng_key(key)
+        noises = random.normal(key, shape=sample_shape + self.event_shape)
```

Comment:
I guess we want `shape=sample_shape + (self.t.shape[-1] - 1,)`

Reply:
implemented as you wrote.